WordPress/Classes/ViewRelated/Comments/CommentsViewController.m

```diff
@@ -1,155 +1,155 @@
-#import "NewCommentsTableViewCell.h"
+#import "CommentsTableViewCell.h"
 #import "CommentViewController.h"
-#import "WordPressAppDelegate.h"
-#import "ReachabilityUtils.h"
-#import "UIColor+Helpers.h"
-#import "WPTableViewSectionHeaderView.h"
+#import "CommentService.h"
 #import "Comment.h"
+#import "Blog.h"
+
+#import "WordPress-Swift.h"
+#import "WPTableViewHandler.h"
+#import "WPGUIConstants.h"
+#import "WPNoResultsView.h"
+#import "UIView+Subviews.h"
 #import "ContextManager.h"
-#import "CommentService.h"
 
-@interface CommentsViewController ()
 
-@property (nonatomic,strong) NSIndexPath *currentIndexPath;
-@property (nonatomic,assign) BOOL moreCommentsAvailable;
+CGFloat const CommentsStandardOffset        = 16.0;
+CGFloat const CommentsSectionHeaderHeight   = 24.0;
+CGRect const CommentsActivityFooterFrame    = {0.0, 0.0, 30.0, 30.0};
+CGFloat const CommentsActivityFooterHeight  = 50.0;
+
+
+@interface CommentsViewController () <WPTableViewHandlerDelegate, WPContentSyncHelperDelegate>
+@property (nonatomic, strong) WPTableViewHandler        *tableViewHandler;
+@property (nonatomic, strong) WPContentSyncHelper       *syncHelper;
+@property (nonatomic, strong) WPNoResultsView           *noResultsView;
+@property (nonatomic, strong) UIActivityIndicatorView   *footerActivityIndicator;
+@property (nonatomic, strong) UIView                    *footerView;
 @end
 
-@implementation CommentsViewController
 
-CGFloat const CommentsStandardOffset = 16.0;
-CGFloat const CommentsSectionHeaderHeight = 24.0;
+@implementation CommentsViewController
 
 - (void)dealloc
 {
-    DDLogMethod();
     [[NSNotificationCenter defaultCenter] removeObserver:self];
-}
-
-- (NSString *)noResultsTitleText
-{
-    return NSLocalizedString(@"No comments yet", @"Displayed when the user pulls up the comments view and they have no comments");
+    _syncHelper.delegate = nil;
+    _tableViewHandler.delegate = nil;
 }
 
 - (void)viewDidLoad
-{
-    DDLogMethod();
-
+{    
     [super viewDidLoad];
-    self.moreCommentsAvailable = YES;
-    self.infiniteScrollEnabled = YES;
-    self.incrementalLoadingSupported = YES;
-    self.title = NSLocalizedString(@"Comments", @"");
-    [WPStyleGuide configureColorsForView:self.view andTableView:self.tableView];
-
-    self.tableView.accessibilityLabel = @"Comments Table";       // required for UIAutomation for iOS 4
-    if ([self.tableView respondsToSelector:@selector(setAccessibilityIdentifier:)]){
-        self.tableView.accessibilityIdentifier = @"Comments Table";  // required for UIAutomation for iOS 5
-    }
-
-    self.editButtonItem.enabled = [[self.resultsController fetchedObjects] count] > 0;
+    
+    [self configureNavBar];
+    [self configureLoadMoreSpinner];
+    [self configureNoResultsView];
+    [self configureRefreshControl];
+    [self configureSyncHelper];
+    [self configureTableView];
+    [self configureTableViewFooter];
+    [self configureTableViewHandler];
+    
+    [self refreshAndSync];
```

Comment:
Thinking about this a bit... In the reader, it makes sense to fetch comments fresh everytime its comment list loads because they are post centric and we always expect different results. But I think in a blog's Comments list, we expect the content to change less frequently. The previous implementation included a mechanism for automatically refreshing only "if needed," which basically meant at least 5 minutes after the last time the list was synced.  There is a similar implementation in the current version of the Post and Page lists.  Would it make sense to retain something along the same lines here?